### PR TITLE
.github: add test coverage with threshold

### PR DESCRIPTION
**Pull request by JochenSiegWork**
- Execute the unittests with the coverage module instead
      of using the unittests module directly.
    - Introduce a threshold for code coverage. If the coverage
      will shrink below the threshold the CI/CD pipeline will
      fail.
    - Collect coverage separately for the core test suite and
      extra packages like chemprop.

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -177,10 +177,14 @@ jobs:
...
           python -m pip install --upgrade pip
+          pip install coverage
           pip install .
       - name: Run unit-tests
         run: |
-          python -m unittest discover -v -s tests -t .
+          # Run only the core test suite in the tests directory.
+          coverage run -m unittest discover tests
+          # Create a coverage report. Fail if the coverage is below 85%. Exclude extra packages from the report.
+          coverage report --fail-under=85 --omit="*chemprop*","*/*chemprop*/*"
```
Comment (c-w-feldmann) [inline .github/workflows/linting.yml:187]: Why is both ("*chemprop*","*/*chemprop*/*") necessary?

**Review by c-w-feldmann — APPROVED** [review]
Just a minor question.

File: .github/workflows/linting.yml (hunk 1 above)
Comment (JochenSiegWork) [reply .github/workflows/linting.yml:187]: The first one matches files with chemprop in their name, and the second one matches directories with chemprop in their name.

Particularly these means:
`"*chemprop*"` matches files like `molpipeline/mol2any/mol2chemprop.py`
`"*/*chemprop*/*"` matches the `molpipeline/estimators/chemprop/` directory (and the contained files and those in subdirectories).

Comment (c-w-feldmann) [reply .github/workflows/linting.yml:187]: ah, I forgot about mol2chemprop

**Pull request merged**